### PR TITLE
Added autohosts

### PR DESCRIPTION
**Pull request by DeinFreund**
Autohosts can be created by an admin by calling !autohost in an existing room.
Autohosts have a fixed title, player limit, password and modoptions.
Autohosts change the map to a random featured map after each battle.
Additionally they can be customized using five new commands:
!maxelo, !minelo, !maxlevel, !minlevel, !minmapsupportlevel

### CODE REVIEW

File: ZkLobbyServer/autohost/Commands/CmdAutohost.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        private string engine;
+        public override string Help => "changes this host into an autohost or sets you as founder";
+        public override string Shortcut => "autohost";
+        public override AccessType Access => AccessType.Admin;
+
+        public override BattleCommand Create() => new CmdAutohost();
+
+        public override string Arm(ServerBattle battle, Say e, string arguments = null)
+        {
+            if (battle.IsMatchMakerBattle)
```
Comment (sprunk) [inline ZkLobbyServer/autohost/Commands/CmdAutohost.cs]: or tourney?

File: ZkLobbyServer/autohost/Commands/BattleCommand.cs [hunk 2]
```diff
@@ -120,6 +120,8 @@ public virtual RunPermission GetRunPermissions(ServerBattle battle, string userN
                 count = battle.Users.Count(x => !x.Value.IsSpectator);
             }
 
+            if (Access == AccessType.Admin && hasAdminRights) return RunPermission.Run;
```
Comment (sprunk) [inline ZkLobbyServer/autohost/Commands/BattleCommand.cs]: battle founder has admin rights so now random people can spawn autohost instances at will (and retain rights as the founder)

**Comment by sprunk** [comment]
appears decent

**Comment by Anarchid** [comment]
Can you please add a lock map option so that i can abuse this to create a hotstepper zombies autohost.

**Pull request merged**